platform-sdk/swirlds-merkle/src/test/java/com/swirlds/virtual/merkle/reconnect/BreakableDataSource.java

```diff
@@ -52,12 +53,16 @@ public void saveRecords(
         final List<VirtualLeafRecord<TestKey, TestValue>> leaves = leafRecordsToAddOrUpdate.toList();
 
         if (builder.numTimesBroken < builder.numTimesToBreak) {
```

Comment:
This is not performance critical code, so let's avoid double-check (anti-)pattern. Actually, no synchronization is needed here at all, if all BrokenBuilder fields are volatile